lsystems/turtle.py

```diff
@@ -70,6 +65,9 @@ def __init__(self, angle, stride, size, width=3):
             "L": Rotation(angle=angle),
             "R": Rotation(angle=-angle),
             ")": Arc(angle=angle, rx=stride, ry=stride),
-            "(": Arc(angle=-angle, rx=stride, ry=stride)
+            "(": Arc(angle=-angle, rx=stride, ry=stride),
+            "[": PushPosition(stack=stack),
+            "]": PopPosition(stack=stack)
+            # TODO: we can extend this to other brackets like "{" with giving PushPosition an argument
```

Comment:
"TODO" can be deleted -- I'll create an issue instead (is a bit cleaner than having TODOs scattered in the code ;))

Reply:
Actually, with the current solution, this is already implemented. Just use another variable like stack2 and add to more symbols with PushPosition and PopPosition using this other stack.